pkg/catalog/routes.go

```diff
@@ -22,20 +22,37 @@ var wildCardRouteMatch trafficpolicy.HTTPRouteMatch = trafficpolicy.HTTPRouteMat
 	Methods:   []string{constants.WildcardHTTPMethod},
 }
 
-// ListOutboundTrafficPolicies returns all outbound traffic policies related to the given service account
-func (mc *MeshCatalog) ListOutboundTrafficPolicies(downstreamIdentity service.K8sServiceAccount) []*trafficpolicy.OutboundTrafficPolicy {
-	outbound := mc.listOutboundPoliciesForTrafficTargets(downstreamIdentity)
+// ListOutboundTrafficPolicies returns all outbound traffic policies
+// 1. from service discovery for permissive mode
+// 2. for the given service account from SMI Traffic Target and Traffic Split
+func (mc *MeshCatalog) ListOutboundTrafficPolicies(downstreamIdentity service.K8sServiceAccount, downstreamServices []service.MeshService) []*trafficpolicy.OutboundTrafficPolicy {
```

Comment:
We should not need downstream services to build the outbound policy. This is a very key requirement to fix #2064.

Reply:
The downstream services will only be used for permissive mode and not with SMI. Further if there's no service it will not impact anything. There are unit tests to reflect the same

Reply:
There's no behavior change from what existed previously, permissive mode always needed the downstream services and was handling the scenario of when there's no service as well like we discussed in the past

Reply:
AS discussed offline, I have removed downstream services from outbound policy